src/java/org/apache/nutch/indexer/IndexerMapReduce.java

```diff
@@ -52,14 +52,34 @@
 import org.apache.nutch.protocol.Content;
 import org.apache.nutch.scoring.ScoringFilterException;
 import org.apache.nutch.scoring.ScoringFilters;
-
-public class IndexerMapReduce extends Configured implements
-    Mapper<Text, Writable, Text, NutchWritable>,
-    Reducer<Text, NutchWritable, Text, NutchIndexAction> {
+import org.apache.nutch.util.NutchConfiguration;
+
+/**
+ * <p>This class is typically invoked from within 
+ * {@link org.apache.nutch.indexer.IndexingJob}
+ * and handles all MapReduce functionality required
+ * when undertaking indexing.</p>
+ * <p>This is a consequence of one or more indexing plugins 
+ * being invoked which extend 
+ * {@link org.apache.nutch.indexer.IndexWriter}.</p>
+ * <p>See 
+ * {@link org.apache.nutch.indexer.IndexerMapReduce#initMRJob(Path, Path, Collection, JobConf, boolean)}
+ * for details on the specific data structures and parameters required for indexing.</p>
+ *
+ */
+public class IndexerMapReduce {
 
   public static final Logger LOG = LoggerFactory
       .getLogger(IndexerMapReduce.class);
 
+  // using normalizers and/or filters
+  private static boolean normalize = false;
+  private static boolean filter = false;
+
+  // url normalizers, filters and job configuration
+  private static URLNormalizers urlNormalizers;
+  private static URLFilters urlFilters;
```

Comment:
No, this would duplicate the variables. Maybe pull out the nested static classes, see this trial https://github.com/sebastian-nagel/nutch/tree/NUTCH-2184